homeassistant/components/geocaching/sensor.py

```diff
@@ -0,0 +1,78 @@
+"""Platform for sensor integration."""
+from __future__ import annotations
+
+import logging
+
+from homeassistant.components.geocaching.coordinator import (
+    GeocachingDataUpdateCoordinator,
+)
+from homeassistant.components.sensor import SensorEntity
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import ATTR_ICON
+
+# from homeassistant.helpers.entity import Entity
+from homeassistant.helpers.typing import HomeAssistantType
+
+from .const import (
+    ATTR_DEFAULT_ENABLED,
+    ATTR_DEVICE_CLASS,
+    ATTR_NAME,
+    ATTR_SECTION,
+    ATTR_STATE,
+    ATTR_UNIT_OF_MEASUREMENT,
+    DOMAIN,
+    SENSOR_ENTITIES,
+)
+from .models import GeocachingEntity
+
+_LOGGER = logging.getLogger(__name__)
+
+
+async def async_setup_entry(
+    hass: HomeAssistantType, entry: ConfigEntry, async_add_entities
+) -> None:
+    """Set up a Geocaching sensor entry."""
+
+    coordinator = hass.data[DOMAIN][entry.entry_id]["COORD"]
+    sensors = [GeocachingSensor(coordinator, key=key) for key in SENSOR_ENTITIES]
+    async_add_entities(sensors, True)
+
+
+class GeocachingSensor(GeocachingEntity, SensorEntity):
+    """Representation of a Sensor."""
+
+    def __init__(
+        self, coordinator: GeocachingDataUpdateCoordinator, *, key: str
+    ) -> None:
+        """Initialize the Geocaching sensor."""
+        self.key = key
+
+        super().__init__(
+            coordinator,
+            enabled_default=bool(SENSOR_ENTITIES[key][ATTR_DEFAULT_ENABLED]),
```

Comment:
```suggestion
            enabled_default=SENSOR_ENTITIES[key][ATTR_DEFAULT_ENABLED],
```

Reply:
If I remove the cast I get the following error in mypy
```
homeassistant/components/geocaching/sensor.py:48: error: Argument "enabled_default" to "__init__" of "GeocachingEntity" has incompatible type "Union[bool, str, None]"; expected "bool"  [arg-type]
```